DataProxy: Add additional settings for dataproxy to help with network proxy timeouts (#27841)
grafana/grafana: docs/sources/administration/configuration.md

This setting also applies to core backend HTTP data sources where query requests use an HTTP client with timeout set.

### keep_alive

Interval between sends of TCP keepalive requests. Default is 30 seconds.

### idle_conn_timeout

> The length of time that idle connections will be maintained. Default is 90 seconds.

Reviewer: ```suggestion
The length of time that Grafana maintains idle connections. Default is 90 seconds.
```

This setting also applies to core backend HTTP data sources where query requests use an HTTP client with timeout set.

### keep_alive_seconds

Interval between keep-alive probes. Default is `30` seconds. For more details check the [Dialer.KeepAlive](https://golang.org/pkg/net/#Dialer.KeepAlive) documentation.

### tls_handshake_timeout_seconds

> The length of time that Grafana will wait for a succussful TLS handshake with the datasource. Default is `10` seconds. For more details check the [Transport.TLSHandshakeTimeout](https://golang.org/pkg/net/http/#Transport.TLSHandshakeTimeout) documentation.

Reviewer 2: ```suggestion
The length of time that Grafana will wait for a successful TLS handshake with the datasource. Default is `10` seconds. For more details check the [Transport.TLSHandshakeTimeout](https://golang.org/pkg/net/http/#Transport.TLSHandshakeTimeout) documentation.
```

### expect_continue_timeout_seconds

> The length of time that Grafana will wait for response headers after initial request headers are written and the request contains the header `Expect: 100-continue`. A value of 0 will result in the body being sent immediately. Default is 1 second.

Reviewer 3: one, not 1

Reviewer 2: In most similar cases in the [configuration](https://grafana.com/docs/grafana/latest/administration/configuration) for the default values we have used the number but enclosed it in backquotes (eg. `1`)
So I suggest to change it like this, unless we have recently decided to migrate away from this and I'm not aware about it.

Author: This was changed in one of the other suggested changes
